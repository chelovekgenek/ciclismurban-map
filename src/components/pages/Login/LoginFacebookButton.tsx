import React, { useState, useCallback } from "react"

import FacebookLogin from "react-facebook-login/dist/facebook-login-render-props"

import { IDispatchProps } from "./LoginFacebookButton.container"
import * as LoginStyles from "./Login.styled"

interface IProps extends IDispatchProps {}

export const LoginFacebookButton: React.FC<IProps> = ({ login }) => {
  const [disabled, setDisabled] = useState(false)
  const onSucess = useCallback(r => r.accessToken && login(r.accessToken), [login])
  const handleFailure = useCallback(() => setDisabled(true), [])
  return (
    <FacebookLogin
      appId={process.env.REACT_APP_FACEBOOK_OAUTH_KEY as string}
      autoLoad
      render={(renderProps: any) => (
        <LoginStyles.Button icon="facebook" onClick={renderProps.onClick} disabled={renderProps.disabled || disabled}>
          Facebook
        </LoginStyles.Button>
      )}
      callback={onSucess}
      onFailure={handleFailure}
    />
  )
}
