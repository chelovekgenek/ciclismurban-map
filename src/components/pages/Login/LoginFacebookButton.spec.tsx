import React from "react"
import { shallow } from "enzyme"

import { LoginFacebookButton } from "./LoginFacebookButton"

describe("LoginFacebookButton", () => {
  const render = (props: Partial<React.ComponentProps<typeof LoginFacebookButton>> = {}) =>
    shallow(<LoginFacebookButton login={jest.fn() as any} {...props} />)
  it("should match snapshot", () => {
    expect(render()).toMatchSnapshot()
  })
})
