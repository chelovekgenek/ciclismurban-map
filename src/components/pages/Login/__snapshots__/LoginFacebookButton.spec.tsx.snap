// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFacebookButton should match snapshot 1`] = `
<t
  appId="735813363592084"
  authType=""
  autoLoad={false}
  callback={[Function]}
  cookie={false}
  disableMobileRedirect={false}
  fields="name"
  isMobile={false}
  language="en_US"
  onFailure={[Function]}
  redirectUri="http://localhost/"
  render={[Function]}
  responseType="code"
  returnScopes={false}
  scope="public_profile,email"
  state="facebookdirect"
  version="2.3"
  xfbml={false}
/>
`;
